### ISSUE TITLE
prepare all workstations: Make sure that it does not create a job for a workstations that already has a job

### ISSUE THREAD

**Issue by urapadmin** [arch-kiosk/arch-kiosk-office#2846]
I had the feeling I saw exactly that in arch1900

**Comment by urapadmin** [comment]
- see #2838 where synchronization ran twice in very quick succession and appeared only once in the even log. 
Is it possible that MCP has an issue?

**Comment by urapadmin** [comment]
in fact if there is already a certain kind of job running (the question here is what is a "certain kind") it should not be possible to queue it again in MCP. This is not limited to preparing a workstation, if goes for synchronization as well.

**Comment by urapadmin** [comment]
this is indeed an issue. I can start two synchronization runs from different computers. They run sequentially, so no harm is done, but it should indeed not even be possible to queue a job like that.

**Comment by urapadmin** [comment]
what really should not happen: a forknexport although a workstation is in the field. 
But apparently the worker has no issues with that. That's a bug.
- This is handled in #3024

**Comment by urapadmin** [comment]
yup, the state machine of the workstation actually allowed a transition from IN_THE_FIELD to fork or export. Not anymore.
But the result is now an ugly 
> StateMachineException('Call to StateMachine.get_transition with unknown transition: FORK')

And the state of the file transfer is gone (I downloaded this workstation and now it says 'no file transfer, yet', which is wrong.

**Comment by urapadmin** [comment]
### Kiosk 1.6.14
- not allowed anymore to queue a job for a workstation that already has a job. Get's an ugly error: 
![image](https://github.com/user-attachments/assets/e276300a-c787-4750-8ef5-10c65355a7f0)

The trick to start two jobs for the same dock is described in #3024

**Comment by luizaogs** [comment]
I didn't get an error but I don't think I'm testing this correctly. 

I prepared my Mac workstation and while that was running hit "prepare all". My Mac finished preparing and doesn't have another job lined up, and the command only started preparing workstations that weren't in the field. This seems as intended - but I'm confused because I see no error. Should I be testing this another way?

**Comment by urapadmin** [comment]
I should have changed the issue's title in the end: I can't find a problem with prepare all. The problem I found was that using the trick in #3024 to start a second job for a dock that just started a job was possible. Now you get that error described above when you try that. You could achieve the same with prepare all, I think: With a dock being in IDLE open the options and leave them open. Now start "prepare all" in a second browser, which will issue a job for that dock, too. Then return to the open options and start "prepare for field". You should get the cryptic Exception promised above.

**Comment by luizaogs** [comment]
I did what you described above with prepare all in two browsers and got a different error, is that fine? 

<img width="1480" alt="Screenshot 2025-01-05 at 5 44 00 PM" src="https://github.com/user-attachments/assets/d03272c3-3811-416a-ad02-4ea069e7015c" />

**Comment by urapadmin** [comment]
hm. No, not what I expected, so I better look into it...

**Comment by urapadmin** [comment]
I get:
![image](https://github.com/user-attachments/assets/816772ca-794a-44ce-9d82-fe67fc29f824)

And I have tried that with both, the dock that is processed first by "prepare all" and with one that is waiting. I was no able to trigger the exception above. But in the end I can live with everything except that a dock that already has a job starts another one. So I don't dig any deeper here.
